compression.go

```diff
@@ -73,6 +85,13 @@ func (w *flateWrapper) Write(p []byte) (int, error) {
 }
 
 func (w *flateWrapper) Close() error {
+	if w.fw == nil {
+		return errors.New("already closed")
+	}
+	defer func() {
+		flateWriterPool.Put(w.fw)
+		w.fw = nil
+	}()
 	err1 := w.fw.Flush()
 	if w.tw.p != [4]byte{0, 0, 0xff, 0xff} {
```

Comment:
Remove the defer and add the following after Flush()

     flatWritePool.Put(w.fw)
     w.fw = nil